sites/public/src/lib/listings/listing-service.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  CombinedListingFilterParams,
+  EnumListingFilterParamsStatus,
+  Listing,
+  OrderByFieldsEnum,
+  OrderParam,
+  PaginatedListing,
+} from "@bloom-housing/backend-core"
+import axios from "axios"
+import qs from "qs"
+import { ListingQueryBuilder } from "./listing-query-builder"
+
+/**
+ * Consolidates listing API calls into a single class
+ * This class is client-side safe
+ */
+export class ListingService {
+  listingsEndpoint: string
+  searchEndpoint: string
+
+  /**
+   * The constructor expects a full URL to the listings endpoint.  This should
+   * usually come from runtimeConfig.getListingServiceUrl().  Note that while
+   * this class is client-side safe, runtimeConfig is not.
+   *
+   * @param listingsEndpoint
+   */
+  constructor(listingsEndpoint: string) {
+    this.listingsEndpoint = listingsEndpoint
+    this.searchEndpoint = listingsEndpoint + "/combined"
```

Comment:
This is why we only need the root.  I figured that the only circumstance that would require this to be anything different would necessitate some major code changes anyway, so changing one or two more lines would be preferable to adding even more env vars